macros/contexts/contextUnits.pl

```diff
@@ -0,0 +1,2056 @@
+
+    Context()->constants->names;
+
+The constants that are units have the C<isUnit> property set.  So
+
+    grep {Context()->constants->get($_)->{isUnit}} (Context()->constants->names);
+
+will get the list of units.
+
+
+=head2 Adding units to other contexts
+
+The C<Units> and C<LimitedUnits> contexts are based on the C<Numeric>
+and C<LimitedNumeric> contexts.  You can add units to other contexts
+using the C<context::Units::extends()> function.  For example,
+
+    Context(context::Units::extending("Fraction")->withUnitsFor("length"));
+
+would allow you to use fractions with units.
+
+In addition to the name of the context to extend, you can pass options
+to C<context::Units::extending()>, as in
+
+    $context = Context(context::Units::extending("LimitedFraction", limited => 1));
+    $context->addUnitsFor("length");
+
+In this case, the C<limited => 1> option indicates that no operations
+are allowed between numbers with units, and since the
+C<LimitedFraction> context doesn't allow operations otherwise, you
+will only be able to enter fractions or whole numbers, with or without
+units, or a unit without a number.
+
+The available options and their defaults are
+
+    keepNegativePowers => 1,     Preserve use of negative powers so C<m s^-1> will
+                                 not be shown as C<m/s> (but will still match it).
+    useNegativePowers => 0       Always use negative powers instead of fractions?
+    limited => 0                 Don't allow operations on numbers with units.
+    exactUnits => 0              Require student units to exactly match correct ones
+                                   in both order and use of negative powers
+    sameUnits => 0               Require student units to match correct ones
+                                   not scaled versions
+    partialCredit => .5          Partial credit if answer is right but units
+    factorUnits => 1             Factor the units out of sums and differences of
+                                   formulas with the same units
+
+The first two and last three can also be set as context flags after
+the context is created.  There is a C<limitedOperators> flag that is
+set by the C<limited> option that controls whether operations are
+allowed on numbers with units, but if you set it, you might also need
+to do
+
+    Context()->parens->set( '(' => { close => ')', type => 'Units' } );
+
+to allow parentheses around units if the parentheses have been removed
+from the original context (as they are in the C<LimitedNumeric>
+context, for instance). This makes it possible to enter units of the
+form C<kg/(m s)> in such contexts.
+
+
+=head2 Creating unit and number-with-unit objects
+
+In the units contexts, units are first-class citizens, and unit and
+number-with-unit objects can be created just like any other
+MathObject.  So you can use
+
+    $n = Compute("3 m/s");
+
+to get a numer-with-units object for 3 meters per second.  You can also use
```

Comment:
typo 'number-with-units'

Reply:
Fixed.  Thanks for the close reading of all of this.